core-services/src/main/java/org/zalando/nakadi/service/publishing/EventOwnerExtractor.java

```diff
@@ -0,0 +1,72 @@
+package org.zalando.nakadi.service.publishing;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.BatchItem;
+import org.zalando.nakadi.domain.EventOwnerHeader;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.Feature;
+import org.zalando.nakadi.exceptions.runtime.JsonPathAccessException;
+import org.zalando.nakadi.service.FeatureToggleService;
+import org.zalando.nakadi.util.JsonPathAccess;
+import org.zalando.nakadi.view.EventOwnerSelector;
+
+import java.util.Collection;
+import java.util.function.Function;
+
+@Component
+public class EventOwnerExtractor {
+
+    private final FeatureToggleService featureToggleService;
+
+    @Autowired
+    public EventOwnerExtractor(final FeatureToggleService featureToggleService) {
+        this.featureToggleService = featureToggleService;
+    }
+
+    public void extract(final Collection<BatchItem> items, final EventType eventType) {
+        final EventOwnerSelector selector = eventType.getEventOwnerSelector();
+        if (null == selector) {
```

Comment:
shouldn't it be set to some default value instead? it looks like events without any header are accessible for everyone.

Reply:
@adyach It should, but it's part of other task. Here we are introducing header extraction